org.eclipse.jgit.pgm/src/org/eclipse/jgit/pgm/Commit.java

```diff
@@ -82,11 +87,17 @@
 	protected void run() throws NoHeadException, NoMessageException,
 			ConcurrentRefUpdateException, JGitInternalException, Exception {
 		try (Git git = new Git(db)) {
+			StoredConfig config = git.getRepository().getConfig();
 			CommitCommand commitCmd = git.commit();
 			if (author != null)
 				commitCmd.setAuthor(RawParseUtils.parsePersonIdent(author));
 			if (message != null)
 				commitCmd.setMessage(message);
+			if (config.getBoolean("commit", null, "gpgsign", false) != false) { //$NON-NLS-1$//$NON-NLS-2$
```

Comment:
I don't think this should be done here but moved to `CommitCommand`.  The JPM wrapper may just invoke a setter there. We may be able to introduce a nice setter there that allows dealing with both options at once.